contracts/Chain.sol

```diff
@@ -126,23 +45,19 @@ contract Chain is BaseChain {
         uint8[] memory _v,
         bytes32[] memory _r,
         bytes32[] memory _s
-    ) public { // it could be external, but for external we got stack too deep
-        uint32 lastBlockId = getLatestBlockId();
-        uint32 lastDataTimestamp = squashedRoots[lastBlockId].extractTimestamp();
+    ) external {
+        // below two checks are only for pretty errors, so we can safe gas and allow for raw revert
```

Comment:
I wonder if we should  remove all the commented code from contracts

Reply:
we agree to remove in next release